src/client/providers/execInTerminalProvider.ts

```diff
@@ -38,8 +40,16 @@ function execInTerminal(fileUri?: vscode.Uri) {
         filePath = `"${filePath}"`;
     }
     const terminal = vscode.window.createTerminal(`Python`);
-    terminal.sendText(`${currentPythonPath} ${filePath}`);
-
+    if (pythonSettings.terminal && pythonSettings.terminal.executeInFileDir) {
+        const fileDirPath = path.dirname(filePath).substring(1);
+        if (fileDirPath != vscode.workspace.rootPath) {
+            terminal.sendText(`cd "${fileDirPath}"`);
+        }
+    }
+    const launchArgs = settings.PythonSettings.getInstance().terminal.launchArgs;
+    const launchArgsString = launchArgs.length > 0 ? " ".concat(launchArgs.join(" ")) : "";
+    terminal.sendText(`${currentPythonPath} ${filePath}${launchArgsString}`);
```

Comment:
Based on #321, shouldn't the order of the command be as follows?
As we're trying to pass arguments into the python interpreter and not the program.
```typescript
terminal.sendText(`${currentPythonPath}${launchArgsString} ${filePath}`);
```